newrelic-agent/src/main/java/com/newrelic/agent/cloud/CloudAccountInfoCache.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *  * Copyright 2024 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package com.newrelic.agent.cloud;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.newrelic.agent.MetricNames;
+import com.newrelic.agent.config.AgentConfig;
+import com.newrelic.agent.service.ServiceFactory;
+import com.newrelic.api.agent.CloudAccountInfo;
+import com.newrelic.api.agent.NewRelic;
+
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.logging.Level;
+
+/**
+ * This class implements the account info methods from the Cloud API.
+ */
+public class CloudAccountInfoCache {
+    private final LoadingCache<Object, Map<CloudAccountInfo, String>> cache;
+    // this object is used to store data that is not related to a specific sdk client
+    private static final Object NULL_CLIENT = new Object();
+
+    CloudAccountInfoCache() {
+        cache = Caffeine.newBuilder()
+                .initialCapacity(4)
+                .weakKeys()
+                .executor(Runnable::run)
+                .build((key) -> Collections.synchronizedMap(new EnumMap<>(CloudAccountInfo.class)));
+    }
+
+    public void setAccountInfo(CloudAccountInfo cloudAccountInfo, String value) {
+        setAccountInfo(NULL_CLIENT, cloudAccountInfo, value);
+    }
+
+    public void setAccountInfo(Object sdkClient, CloudAccountInfo cloudAccountInfo, String value) {
+        if (sdkClient == null) {
+            return;
+        }
+        if (value == null) {
+            Map<CloudAccountInfo, String> accountInfo = cache.getIfPresent(sdkClient);
+            if (accountInfo != null) {
+                accountInfo.remove(cloudAccountInfo);
+            }
+        }
```

Comment:
Should there be a short-circuit return here or is it ok to have a `cloudAccountInfo` with a null value?

Reply:
yes